k2/csrc/ragged_ops.cu

```diff
@@ -235,11 +235,18 @@ RaggedShape RaggedShapeFromTotSizes(ContextPtr c, int32_t num_axes,
   NVTX_RANGE(K2_FUNC);
   K2_CHECK_GE(num_axes, 2);
   std::vector<RaggedShapeLayer> axes(num_axes - 1);
-  // In future we might choose to allocate everything in one big array, to avoid
-  // multiple allocations, but for now just do it the simple way.
+  int32_t tot_size = 0;
   for (int32_t axis = 1; axis < num_axes; ++axis) {
-    axes[axis - 1].row_splits = Array1<int32_t>(c, tot_sizes[axis - 1] + 1);
-    axes[axis - 1].row_ids = Array1<int32_t>(c, tot_sizes[axis]);
+    tot_size += tot_sizes[axis - 1] + 1 + tot_sizes[axis];
+  }
+  Array1<int32_t> splits_ids(c, tot_size);
```

Comment:
`splits_ids` is a pre-allocated memory buffer for both `row_splits` and `row_ids`. Maybe it'd better to name it `buf` or `tmp`.